lib/dotcom_web/components/gtfs.ex

```diff
@@ -0,0 +1,188 @@
+defmodule DotcomWeb.Components.Gtfs do
+  @moduledoc """
+  Reusable components for displaying UI derived from GTFS data.
+  """
+  use CVA.Component
+  use Phoenix.Component
+
+  import MbtaMetro.Components.Icon
+
+  alias Routes.Route
+
+  @logan_express_icon_names Route.logan_express_icon_names()
+  @massport_icon_names Route.massport_icon_names()
+
+  variant(
+    :size,
+    [
+      small: "rounded-[3px] px-[2px] py-[2px] text-sm",
+      default: "rounded-[4px] px-[4px] py-[4px]"
+    ],
+    default: :default
+  )
+
+  attr(:class, :string, default: "")
+  attr(:rest, :global)
+  attr(:route, Routes.Route, required: true)
+
+  @doc """
+  Creates a simple symbol representing a single route. Currently shows rounded
+  SVG icons, small pills for bus routes, custom icons for Logan Express and
+  Massport routes. Supports a `size` variant defined as either `:default` or
+  `:small`.
+
+  ```elixir
+  <.route_symbol route={%Routes.Route{id: "Red"}} size="small" />
+  ```
+
+  Supports additional CSS class names, though they may have limited effect. Use
+  with caution.
+
+  ```elixir
+  <.route_symbol route={%Routes.Route{type: 3}} class="mr-3" />
+  ```
+
+  Supports custom HTML attributes.
+
+  ```elixir
+  <.route_symbol route={%Routes.Route{type: 3}} data-toggle="tooltip" />
+  ```
+  """
+  def route_symbol(
+        %{
+          route: %Route{
+            description: description,
+            external_agency_name: nil,
+            type: 3
+          }
+        } = assigns
+      ) do
+    if description in [:rail_replacement_bus, :rapid_transit] do
```

Comment:
💡 Could be done with pattern matching.

Reply:
I agree with this, although per my comment about about possibly using `&Routes.Route.icon_atom/1`, I think the optimal pattern-matching might not be a one-to-one match with the code that's here.